### PR TITLE
Pairtools parse

### PR DESCRIPTION
**Pull request by jianhong**
## PR checklist

- [ x ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ x ] If necessary, include test data in your PR.
- [ x ] Remove all TODO statements.
- [ x ] Emit the `<SOFTWARE>.version.txt` file.
- [ x ] Follow the naming conventions.
- [ x ] Follow the parameters requirements.
- [ x ] Follow the input/output options guidelines.
- [ x ] Add a resource `label`
- [ x ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
    - [ x ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [ x ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd`

### CODE REVIEW

File: tests/software/pairtools/parse/main.nf [hunk 1]
```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env nextflow
+
+nextflow.enable.dsl = 2
+
+process NO_PG {
```
Comment (drpatelh) [inline tests/software/pairtools/parse/main.nf:5]: Same comment here as with `pairtools/flip`

File: software/pairtools/parse/main.nf [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    script:
+    def software = getSoftwareName(task.process)
+    def prefix   = options.suffix ? "${meta.id}${options.suffix}" : "${meta.id}"
+    """
+    pairtools parse -c ${chromsizes} \\
+        --assembly ${params.species} \\
+        $options.args \\
+        --output-stats ${prefix}.raw.pairsam.stat \\
+        -o ${prefix}.raw.pairsam.gz \\
+        ${bam}
```
Comment (drpatelh) [inline software/pairtools/parse/main.nf:39]:
```suggestion
    pairtools \\
        parse \\
        -c $chromsizes \\
        --assembly $species \\
        $options.args \\
        --output-stats ${prefix}.pairsam.stat \\
        -o ${prefix}.pairsam.gz \\
        $bam
```

File: software/pairtools/parse/main.nf [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    tuple val(meta), path("*.pairsam.gz")  , emit: pairsam
+    tuple val(meta), path("*.pairsam.stat"), emit: stat
+    path "*.version.txt"                   , emit: version
+
+    script:
+    def software = getSoftwareName(task.process)
+    def prefix   = options.suffix ? "${meta.id}${options.suffix}" : "${meta.id}"
+    """
+    pairtools parse -c ${chromsizes} \\
+        --assembly ${params.species} \\
```
Comment (drpatelh) [inline software/pairtools/parse/main.nf:35]: Is `--assembly` a mandatory parameter? If it is then maybe we just have another `input` value called `species`?

File: tests/software/pairtools/parse/main.nf [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+      path sam
+    output:
+      path "filtered.file", emit: fil
+    script:
+      """
+      zcat < ${sam} | grep -v "@PG" > filtered.file
+      """
+}
+
+include { PAIRTOOLS_PARSE } from '../../../../software/pairtools/parse/main.nf' addParams( options: [:] )
```
Comment (drpatelh) [inline tests/software/pairtools/parse/main.nf:19]:
```suggestion
include { PAIRTOOLS_PARSE } from '../../../../software/pairtools/parse/main.nf' addParams( options: ['suffix':'raw'] )
```

File: tests/software/pairtools/parse/main.nf [hunk 5]
```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env nextflow
+
+nextflow.enable.dsl = 2
+
+include { PAIRTOOLS_PARSE } from '../../../../software/pairtools/parse/main.nf' addParams( options: ['suffix':'.raw'] )
+
+workflow test_pairtools_parse {
+
+    input = [ [ id:'test', single_end:false ], // meta map
+              file("https://raw.githubusercontent.com/open2c/pairtools/master/tests/data/mock.sam", checkIfExists: true) ]
```
Comment (drpatelh) [inline tests/software/pairtools/parse/main.nf:10]: I have just realised that the module can take BAM input too. Can we not use the test data we have already to get the BAM file like [here](https://github.com/nf-core/modules/blob/5e18e36895ba7a63e864faad4105e14597ffcddb/tests/software/bedtools/genomecov/main.nf#L9) and the sizes file like [here](https://github.com/nf-core/modules/blob/5e18e36895ba7a63e864faad4105e14597ffcddb/tests/software/bedtools/complement/main.nf#L11)

Comment (drpatelh) [reply tests/software/pairtools/parse/main.nf:10]: I will merge all of these in now but be great if you are able to update them all at some point if possible.

**Review by drpatelh — APPROVED** [review]
🥇

File: tests/software/pairtools/parse/main.nf (hunk 5 above)
Comment (jianhong) [reply tests/software/pairtools/parse/main.nf:10]: I will do full test for the bam file. Once I confirmed the BAM file can work correctly, I will update the tool.

**Pull request merged**